tests/returntypes/test_autograd_qnode_new.py

```diff
@@ -0,0 +1,1942 @@
+
+        res = cost_fn(x, y)
+
+        if diff_method not in {"backprop"}:
+            pytest.skip("Test only supports backprop")
+
+        res = qml.jacobian(cost_fn)(x, y)
+        expected = np.array([-np.sin(x) * np.cos(y) / 2, -np.cos(x) * np.sin(y) / 2])
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+    def test_projector(self, dev_name, diff_method, mode, tol):
+        """Test that the variance of a projector is correctly returned"""
+        if diff_method == "adjoint":
+            pytest.skip("Adjoint does not support projectors")
+
+        dev = qml.device(dev_name, wires=2)
+        P = np.array([1], requires_grad=False)
+        x, y = np.array([0.765, -0.654], requires_grad=True)
+
+        @qnode(dev, diff_method=diff_method, interface="autograd", mode=mode)
+        def circuit(x, y):
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=1)
+            qml.CNOT(wires=[0, 1])
+            return qml.var(qml.Projector(P, wires=0) @ qml.PauliX(1))
+
+        res = circuit(x, y)
+        expected = 0.25 * np.sin(x / 2) ** 2 * (3 + np.cos(2 * y) + 2 * np.cos(x) * np.sin(y) ** 2)
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+        res = qml.jacobian(circuit)(x, y)
+        expected = np.array(
+            [
+                [
+                    0.5 * np.sin(x) * (np.cos(x / 2) ** 2 + np.cos(2 * y) * np.sin(x / 2) ** 2),
+                    -2 * np.cos(y) * np.sin(x / 2) ** 4 * np.sin(y),
+                ]
+            ]
+        )
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+
+@pytest.mark.xfail(reason="CV variable needs to be update for the new return types.")
+@pytest.mark.parametrize(
+    "diff_method,kwargs",
+    [["finite-diff", {}], ("parameter-shift", {}), ("parameter-shift", {"force_order2": True})],
+)
+class TestCV:
+    """Tests for CV integration"""
+
+    def test_first_order_observable(self, diff_method, kwargs, tol):
+        """Test variance of a first order CV observable"""
+        dev = qml.device("default.gaussian", wires=1)
+
+        r = np.array(0.543, requires_grad=True)
+        phi = np.array(-0.654, requires_grad=True)
+
+        @qnode(dev, diff_method=diff_method, **kwargs)
+        def circuit(r, phi):
+            qml.Squeezing(r, 0, wires=0)
+            qml.Rotation(phi, wires=0)
+            return qml.var(qml.X(0))
+
+        res = circuit(r, phi)
+        expected = np.exp(2 * r) * np.sin(phi) ** 2 + np.exp(-2 * r) * np.cos(phi) ** 2
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+        # circuit jacobians
+        res = qml.jacobian(circuit)(r, phi)
+        expected = np.array(
+            [
+                [
+                    2 * np.exp(2 * r) * np.sin(phi) ** 2 - 2 * np.exp(-2 * r) * np.cos(phi) ** 2,
+                    2 * np.sinh(2 * r) * np.sin(2 * phi),
+                ]
+            ]
+        )
+        assert np.allclose(res, expected, atol=tol, rtol=0)
```

Comment:
Good to check type & shape - this may actually not yet be as expected, as the CV param-shift transform has not been reworked yet. It may still (by accident) coincide with what we'd expect.

Reply:
I am not updating the CV test and should be done in a separate PR.